File: python/Programmers_level/level1/약수의 합.py
def solution(n):
    result = []
    # 약수찾기
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            result.append(i)
            if n // i not in result:
                result.append(n // i)
                
    answer = sum(result)
    return answer

File: Javascript/Programmers_level/level1/약수의 합.js
function solution(n) {
  let answer = 0
  let arr = []
  for (let i = 1; i < Math.sqrt(n) + 1; i++) {
    if (n % i === 0) {
      arr.push(i, n / i)
    }
  }
  let set = new Set([...arr])
  set.forEach((a) => (answer += a))
  return answer
}

// 조건문으로 약수중복 처리
// function solution(n) {
//     let answer = 0;
//     for (let i  = 1; i < parseInt(Math.sqrt(n)) + 1; i++) {
//         if (n % i === 0) {
//             answer += i
//             if (i === n / i) {
//                 continue
//             }
//             answer += n / i
//         }
//     }
//     return answer;
// }
